packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts

```diff
@@ -54,5 +68,30 @@ export class PluginVscodeCommandsContribution implements CommandContribution {
                 this.contextKeyService.createKey(String(contextKey), contextValue);
             }
         });
+
+        commands.registerCommand(VscodeCommands.PREVIEW_HTML, {
+            isVisible: () => true,
+            // tslint:disable-next-line: no-any
+            execute: (resource: URI, position?: any, label?: string, options?: any) => {
+                label = label || resource.fsPath;
+                const view = new WebviewWidget(label, {allowScripts: true}, {});
+                this.resorces(new TheiaURI(resource)).then(res => {
+                    res.readContents().then(str => {
+                        const html = '<!DOCTYPE html><html><head></head>' + str + '</html>';
+                        this.shell.addWidget(view, { area: 'main', mode: 'split-right' });
+                        this.shell.activateWidget(view.id);
+                        view.setHTML(html);
+                    });
+                    this.editorManager.getOrCreateByUri(new TheiaURI(resource)).then(editor => {
+                        editor.editor.onDocumentContentChanged(listener => {
+                            const html = '<!DOCTYPE html><html><head></head>' + editor.editor.document.getText() + '</html>';
```

Comment:
string literal as well there ?

maybe we could have a method that is called to provide the html based on a given text

Reply:
done